### PR TITLE
feat(webdriver): extract webdriver-manager into a separate node module

### LINKED ISSUE #607: Extract "webdriver-manager" as an independent Node module
The latest project I'm working on is not Angular-based, so using Protractor doesn't make much sense.  However, specific parts of Protractor would still be incredibly useful for other WebDriver purposes.

I'd love to see `webdriver-manager` extracted from Protractor into its own independent Node module. As far as the `webdriver-manager update` process goes, then, you'd probably either (a) want to install it into the `webdriver-manager` module folder and then provide a class that Protractor can `require` to get the path, (b) allow a path to be passed in as an arg, or (c) install into the CWD.

Thoughts?  At a cursory glance, I don't see any strong reason why it can't be easily extracted. Adding a dependency on `protractor` just to get `webdriver-manager` seems a bit extreme.

### LINKED ISSUE #2402: Add ability to create chromedriver.log
I have seen that chromedriver can output a logfile (https://sites.google.com/a/chromium.org/chromedriver/logging)

This page shows how to set this up when executing the exe directly: `chromedriver.exe --verbose --log-path=chromedriver.log`

There is no apparent way to set these arguments from protractor however.

See this StackOverflow question for more info:

http://stackoverflow.com/questions/31662828/how-to-access-chromedriver-logs-for-protractor-test/31662935

### PR DESCRIPTION
**Pull request by cnishina**
closes #607, closes #2402

### CODE REVIEW

**Comment by NickTomlin** [comment]
So much :clap: :confetti_ball: 

This is very OT, but have we considered asking the `webdriver-manager` folks to transfer ownership to the angular org and publishing this there?

**Comment by sjelin** [comment]
Nit: commit should say "extract webdriver-tool into..."

re: @NickTomlin 
Yeah, Craig, if you want I'm sure you could get a repo under `angular`.

Also, we are the `webdriver-manager` folks :smile:

**Comment by juliemr** [comment]
We're discussing with the owner of `webdriver-manager` on npm - conclusion TBD.

**Comment by NickTomlin** [comment]
@sjelin ah yeah I should have been clearer, I mean the community npm [webdriver-manager](https://www.npmjs.com/package/webdriver-manager) fork

Glad to hear we are talking with the NPM owner. Hopefully we can work that out!

**Comment by cnishina** [comment]
Name change from webdriver-tool to [webdriver-manager](https://github.com/angular/webdriver-manager). Updating this pr to reflect this.

**Comment by cnishina** [comment]
browserStack tests are failing which needs to be addressed.

File: README.md [hunk 1]
```diff
@@ -45,9 +45,10 @@ Clone the github repository:
     npm install
     cd ..
 
-Start up a selenium server. By default, the tests expect the selenium server to be running at `http://localhost:4444/wd/hub`. A selenium server can be started with `webdriver-manager`.
+Start up a selenium server. By default, the tests expect the selenium server to be running at `http://localhost:4444/wd/hub`. A selenium server can be started with [webdriver-manager](https://github.com/angular/webdriver-manager).
```
Comment (juliemr) [inline README.md]: Maybe add something like: "which is included in bin/webdriver-manager"?

File: config.json [hunk 2]
```diff
@@ -3,5 +3,11 @@
     "selenium": "2.53.0",
     "chromedriver": "2.21",
     "iedriver": "2.53.1"
+  },
+  "cdnUrls": {
```
Comment (juliemr) [inline config.json]: Are these not the default?

**Comment by juliemr** [comment]
We should mention in the commit description that this will change the filenames by giving everything versions.

**Comment by cnishina** [comment]
* As discussed, versioning of binaries was previously added in ~3.2
* Fixes for webdriver-manager 10.0.4 for configuration file and path issues.
* Removed the config.json. This will be managed now by webdriver-manager
* wedriver-manager downloads the file to the node_modules/webdriver-manager/selenium folder. This will no longer appear in the protractor directory.

**Pull request merged**